components/match2/wikis/starcraft2/legacy/match_maps_legacy_store.lua

```diff
@@ -0,0 +1,100 @@
+---
+-- @Liquipedia
+-- wiki=starcraft2
+-- page=Module:Match maps/LegacyStore
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Match = require('Module:Match')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
+local Template = require('Module:Template')
+local WarningBox = require('Module:WarningBox')
+
+local MatchGroupBase = Lua.import('Module:MatchGroup/Base', {requireDevIfEnabled = true})
+
+local globalVars = PageVariableNamespace()
+local matchlistVars = PageVariableNamespace('LegacyMatchlist')
+
+local MatchMapsLegacyStore = {}
+
+function MatchMapsLegacyStore.init(frame)
+	local args = Arguments.getArgs(frame)
+	return MatchMapsLegacyStore._init(args)
+end
+
+function MatchMapsLegacyStore._init(args)
+	local store = Logic.nilOr(
+		Logic.readBoolOrNil(args.store),
+		not Logic.readBool(globalVars:get('disable_SMW_storage'))
+	)
+
+	local warnings = {}
+	table.insert(warnings, 'This is a legacy matchlist! Please use the new matchlist instead.')
+
+	if store then
+		local bracketId = MatchGroupBase.readBracketId(args.id)
+		table.insert(warnings, MatchGroupBase._checkBracketDuplicate(bracketId) or nil)
+		matchlistVars:set('bracketid', bracketId)
+	end
+
+	matchlistVars:set('matchListTitle', args.title or args[1] or 'Match List')
+	matchlistVars:set('warnings', Json.stringify(warnings))
+	matchlistVars:set('store', store and 'true' or nil)
+end
+
+function MatchMapsLegacyStore.close()
+	local bracketId = matchlistVars:get('bracketid')
+	if not bracketId then
+		return '</table>'
```

Comment:
Should this really be done in the module and not the template? (same with the other /table)

Reply:
nope, overlooked it, can be removed entirely as all 1v1 matchlists have been converted, so the `if not bracketId then` is pretty useless as bracketid should always be set